frame/aura/src/lib.rs

```diff
@@ -180,10 +202,19 @@ impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
 	{
 		// instant changes
 		if changed {
-			let next_authorities = validators.map(|(_, k)| k).collect::<Vec<_>>();
+			let next_authorities = validators
```

Comment:
So we could have a a situation where staking and session are happily tossing around a large number of validators, but in reality a small number of them end up doing something useful and get reward. 

We should def. `warn` or `err` log here in case `.take` actually truncates the validator set.